spec/acceptance/producer_spec.rb

```diff
@@ -1,162 +0,0 @@
-require 'spec_helper_acceptance'
-
-if fact('operatingsystemmajrelease') == '6' && fact('osfamily') == 'RedHat'
-  user_shell = '/bin/bash'
-else
-  case fact('osfamily')
-  when 'RedHat', 'Suse'
-    user_shell = '/sbin/nologin'
-  when 'Debian'
-    user_shell = '/usr/sbin/nologin'
-  end
-end
-
-describe 'kafka::producer', if: (fact('operatingsystemmajrelease') == '6' && fact('osfamily') == 'RedHat') do # systemd systems not supported by kafka::producer::service
```

Comment:
Is it still true that the producer doesn't work on systemd? If you'd keep it in, would it work? If not, what (supported) code would actually call `kafka::producer`?

Reply:
If you would be able to wire up the stdin of the producer with something it would work. Though I do not know how this could work. `kafka::producer` could only be called by a user from the module. Not sure what todo with it.

Reply:
With systemd socket activation I think that might be possible. You can connect a socket to stdin. Perhaps an issue should be opened for that?

If I look through metadata.json, I'm only unsure about SLES but otherwise all distros run systemd. That can suggest to drop the producer code altogether as well.

Reply:
I created this one here: https://github.com/voxpupuli/puppet-kafka/issues/324